plugins/algebra/eigen/CMakeLists.txt

```diff
@@ -25,5 +25,7 @@ target_compile_definitions( detray_eigen
 # though they are set up from a "system include path". So I had to explicitly
 # disable the warning triggered by those headers, for anything that uses them.
 # This is pretty bad, as we're now blind to these types of warnings/errors. :-(
+if ( "${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU" )
```

Comment:
It is supposed to be `if ( "${CMAKE_CUDA_COMPILER_ID}" MATCHES "NVIDIA" )` but somehow `${CMAKE_CUDA_COMPILER_ID}` is not visible here... But should be OK because anyway we can remove this part once Eigen 3.4.1 is released someday

Reply:
Of course what's relevant here is [CMAKE_CUDA_HOST_COMPILER](https://cmake.org/cmake/help/latest/variable/CMAKE_CUDA_HOST_COMPILER.html). But unfortunately CMake does not inspect that compiler. And I think the variable is only set to anything when/if you ask for something explicitly.

I fear that there's no simple solution here. A more elaborate way would be to:
  - Check if `CMAKE_CUDA_HOST_COMPILER` is set;
    * If not, just assume that `CMAKE_CXX_COMPILER_ID` is the one to use. Although it very well may not be.
    * If it is, compare `CMAKE_CXX_COMPILER` with `CMAKE_CUDA_HOST_COMPILER`.
      - If they are the same, you can confidently use `CMAKE_CXX_COMPILER_ID`;
      - If they are different, you would have to do some fragile string comparisons on the path. :frowning:

So all in all, I can just not come up with a bulletproof way of doing it. As whatever I think of, would still have some blind spots. And at that point we might as well just go with something drastically simpler (your current proposal), which has just a very slightly bigger blind spot.